commit 1865bc71a869ede69098b1f3e65857b4330f0607
Author: Borislav Petkov <bp@suse.de>
Date:   Tue Dec 22 18:55:06 2020 +0100

    EDAC/amd64: Limit error injection functionality to supported hw
    
    Families up to and including 0x16 allow access to the injection
    hardware. Starting with family 0x17, access to those registers is
    blocked by security policy.
    
    Limit that only on the families which support it.
    
    Suggested-by: Yazen Ghannam <yazen.ghannam@amd.com>
    Signed-off-by: Borislav Petkov <bp@suse.de>
    Link: https://lkml.kernel.org/r/20201222180013.GD13463@zn.tnic

diff --git a/drivers/edac/amd64_edac.c b/drivers/edac/amd64_edac.c
index d55f8ef2240c..9868f95a5622 100644
--- a/drivers/edac/amd64_edac.c
+++ b/drivers/edac/amd64_edac.c
@@ -828,9 +828,11 @@ static umode_t inj_is_visible(struct kobject *kobj, struct attribute *attr, int
 	struct mem_ctl_info *mci = container_of(dev, struct mem_ctl_info, dev);
 	struct amd64_pvt *pvt = mci->pvt_info;
 
-	if (pvt->fam < 0x10)
-		return 0;
-	return attr->mode;
+	/* Families which have that injection hw */
+	if (pvt->fam >= 0x10 && pvt->fam <= 0x16)
+		return attr->mode;
+
+	return 0;
 }
 
 static const struct attribute_group inj_group = {